### DISCUSSION TITLE
Questions about ARM and Release schedules

### DISCUSSION THREAD

**Discussion by briansturgill** [elalish/manifold#752] [Q&A]
I am nearing first release for Piecad and need to understand where things stand.

Am I correct in understanding that the PyPi wheels and general build checking is done only on Intel machines?
Has it been tried on ARM machines?
I don't have an ARM Windows machine, but I do have an ARM Mac and a Raspberry Pi, and so could try building there
if you'd find that helpful.

You seem to release sporadically. I really need 2d Hull and other fixes to make it into PyPi. Otherwise
I'll have to send out my own version of manifold3d within the Piecad release.
Can you give me some idea about when that might occur?

**Comment by elalish** [comment]
Fair point on sporadic releases. Yes, we should probably do one soon. I've been fighting off a variety of illnesses lately, which has slowed progress. I'll try to get one out in the next week or two. If you need it sooner, let me know and we can discuss more. 

The hardware is pretty much whatever Github Actions uses, which is a bit opaque. Personally, I develop on an M1 Macbook Pro, so that at least is well tested. More testing is always welcome, but if you want to file issues on systems we can't put in our CI, I'd ask you to also be responsible for fixing what you find 😄 - we'll do our best to help, but if you have the only repro device...

**Comment by pca006132** [comment]
I think we usually do a release when someone asked for it, when they need certain things/fixes that are not yet published to PyPi/npm. We don't really have a regular release schedule.

The PyPi wheels do build for `macosx_arm64`, but we don't have automated test for it on the CI because GitHub action does not allow us to choose the runner.

**Comment by briansturgill** [comment]
OK, I can confirm it works under a Raspberry Pi 4.
(Has only two threads of execution, so I took out TBB.)
On my Linux box, the manifold test suite runs in 29,215ms.
On the Raspberry Pi, the test suite runs in 329,524ms.
OpenSCAD does run on RaspberryPis.

**Comment by briansturgill** [comment]
@elalish It seems you are already building for ARM macs... it just installed and I've run "all_apis.py" on both platforms
(sans the 6 missing calls).
